src/printing.jl

```diff
@@ -192,5 +192,468 @@ function show(stream::IO, x::Union{LazySum, LazyProduct})
+                        print(io, undef_ref_str)
+                    else
+                        el = X[mirror_world_index(i, ldims), mirror_world_index(j, rdims)]
+                        show(io, el)
+                    end
+                end
+                if last(cr) == last(indc)
+                    i < last(indr) && print(io, "; ")
+                elseif cdots
+                    print(io, " \u2026 ")
+                end
+            end
+        end
+        last(rr) != nr && rdots && print(io, "\u2026 ; ")
+    end
+    print(io, "]")
+end
+
+
+function showarray_stdord(io::IO, X::AbstractVecOrMat, ldims::Vector, rdims::Vector, repr::Bool = true; header = true)
+    if repr && ndims(X) == 1
+        return show_vector_std(io, X, ldims, "[", "]")
+    end
+    if !haskey(io, :compact)
+        io = IOContext(io, :compact => true)
+    end
+    if !repr && get(io, :limit, false) && eltype(X) === Method
+        # override usual show method for Vector{Method}: don't abbreviate long lists
+        io = IOContext(io, :limit => false)
+    end
+    (!repr && header) && print(io, summary(X))
+    if !isempty(X)
+        (!repr && header) && println(io, ":")
+        if ndims(X) == 0
+            if isassigned(X)
+                return show(io, X[])
+            else
+                return print(io, undef_ref_str)
+            end
+        end
+        if repr
+            print_matrix_repr_std(io, X, ldims, rdims)
+        else
+            punct = (" ", "  ", "")
+            print_matrix_std(io, X, ldims, rdims, punct...)
+        end
+    elseif repr
+        Base.repremptyarray(io, X)
+    end
+end
+showarray_stdord(io::IO, X::Vector, dims::Vector, repr::Bool = true; header = true) = showarray_stdord(io, X, dims, [1], repr; header = header)
+
+
+function showsparsearray_stdord(io::IO, S::SparseMatrixCSC, ldims::Vector, rdims::Vector)
+    if nnz(S) == 0
+        return Base.show(io, MIME("text/plain"), S)
+    end
+
+    limit::Bool = get(io, :limit, false)
+    if limit
+        rows = displaysize(io)[1]
+        half_screen_rows = div(rows - 8, 2)
+    else
+        half_screen_rows = typemax(Int)
+    end
+    pad = ndigits(max(S.m,S.n))
+    k = 0
+    sep = "\n  "
+    if !haskey(io, :compact)
+        io = IOContext(io, :compact => true)
+    end
+    for col = 1:S.n, k = S.colptr[col] : (S.colptr[col+1]-1)
+        if k < half_screen_rows || k > nnz(S)-half_screen_rows
+            print(io, sep, '[',
+                    rpad(mirror_world_index(S.rowval[k], reverse(ldims)), pad),
+                    ", ",
+                    lpad(mirror_world_index(col, reverse(rdims)), pad),
+                    "]  =  ")
+            if isassigned(S.nzval, Int(k))
+                show(io, S.nzval[k])
+            else
+                print(io, Base.undef_ref_str)
+            end
+        elseif k == half_screen_rows
+            print(io, sep, '\u22ee')
+        end
+        k += 1
```

Comment:
Again this line has no effect since `k` is overwritten by the next step in the loop.